assets/dev/js/editor/document/save/behaviors/footer-saver.js

```diff
@@ -39,8 +41,34 @@ module.exports = class FooterSaver extends Marionette.Behavior {
 		this.ui.buttonSaveOptions.toggleClass( 'elementor-disabled', ! hasChanges );
 	}
 
+	activateCopyAndShareButton( postStatus, status ) {
+		if ( ! status ) {
+			return;
+		}
+		if ( 'publish' === postStatus ) {
+			this.ui.copyAndShareLink.removeClass( 'elementor-disabled' );
+		} else {
+			this.ui.copyAndShareLink.addClass( 'elementor-disabled' );
+		}
+	}
+
+	onCopyAndShareLinkClick() {
+		const url = elementor.config.document.urls.permalink ?? '';
+		if ( ! url ) {
+			return;
+		}
+		navigator.clipboard.writeText( url );
+		const $title = this.ui.copyAndShareLink.find( '.elementor-title' );
+		$title.text( 'Link copied!' );
+		setTimeout( () => {
+			$title.text( 'Copy and share Link' );
+		}, 2000 );
+	}
+
 	onRender() {
 		this.addTooltip();
+		const postStatus = elementor.config.document.status.value ?? '';
+		this.activateCopyAndShareButton( postStatus, true );
```

Comment:
Should this not depend on the Document type as well?

Reply:
@louiswol94 not sure because the presence/absence of the button is controlled in the template. here I am just activating / deactivating it based on the post status, but this activation/deactivation is not tied to the button being present, I mean if the button is not here it does nothing. I could probably refactor everything and handle everything inside this class, without touching the template. What do you think? It would be cleaner I think so I feel it's better to refactor.

Reply:
Is it ok if I refactor in a separate PR @louiswol94?

Reply:
@nicoladj77 As long as it doesn't affect the other Document types for now 👍